Get typing tests to work on Python 3.9

The typing tests currently fail on Python 3.9 and mypy 0.770. It's possible that upgrading mypy to the recently-released 0.780 will fix the issue as it has some fixes for 3.9, e.g. https://github.com/python/mypy/pull/8716

### Reproducing code example:
Do a

```
$ ./runtests.py -s tests
```

on Python 3.9.

### Error message:
The tests will have many errors of the form:

```
E               ValueError: Unexpected reveal line format: /home/travis/build/numpy/numpy/builds/venv/lib/python3.9/site-packages/numpy/__init__.pyi:53: error: syntax error in type comment
```

(which seem potentially related to the issue linked above).

### Numpy/Python version information:
- NumPy version: 1.20.0.dev0+f09b27d
- This is on Python 3.9, though I'm relaying errors seen in CI (haven't set up 3.9 locally yet)

--- a/test_requirements.txt
+++ b/test_requirements.txt
@@ -11,5 +11,5 @@ cffi
 # - Mypy relies on C API features not present in PyPy
 # - Mypy doesn't currently work on Python 3.9
 # - Python 3.6 doesn't work because it doesn't understand py.typed
-mypy==0.770; platform_python_implementation != "PyPy" and python_version > "3.6" and python_version < "3.9"
+mypy==0.780; platform_python_implementation != "PyPy" and python_version > "3.6"
 typing_extensions

